spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java

```diff
@@ -1046,6 +1046,16 @@ public void setMissingTopicsFatal(boolean missingTopicsFatal) {
 		 */
 		private String keyPassword;
 
+		/**
+		 * Certificate chain in the format specified by 'ssl.keystore.type'.
```

Comment:
My understanding from the KIP and Kafka docs is that only PEM is supported; I wonder if we should indicate that here and/or enforce it in `buildProperties()`?

OTOH, leaving it like this will make it future proof.

Reply:
I had copied [the comments from the SslConfigs source](https://github.com/apache/kafka/blob/e8754ba7a01f33ba503ef79de66a5f644cc2ced8/clients/src/main/java/org/apache/kafka/common/config/SslConfigs.java#L76).
I had left out the lines immediately following it that detail (current) support for formats as a way of future proofing.

I'm fine with any which way: expanding the comments here, refer to SslConfigs in JavaDoc, or leaving it as is.
This comment is mostly to provide context; up to you how to best document this going forward.

Reply:
We could also do with some tests for the new support. That would help us to make an informed decision around what to do when the type is and is not set and when it's set to a contradictory value.